spec/controllers/internal_api/v1/pillars_controller_spec.rb

```diff
@@ -434,97 +434,70 @@
+            email:    "mail",
+            name:     "name"
+          }
+        },
+        group:           {
+          base_dn:  "cn=groups,#{dn_common}",
+          filter:   "(objectClass=group)",
+          attr_map: {
+            user:  "uid",
+            group: "member",
+            name:  "name"
+          }
         }
       }
-    }
-  end
-
-  # rubocop:disable RSpec/ExampleLength
-  context "with dex LDAP connectors tls" do
-    it "has dex LDAP connectors" do
-      dex_connector_ldap = create(:dex_connector_ldap, :tls, :regular_admin)
-      CertificateService.create(service: dex_connector_ldap, certificate: certificate)
-
-      expected_json = {
+      connector_json = connector_json.merge(bind: { anonymous: true }) if conn.bind_anon
+      {
         registries:          [],
         kubelet:             {
           :"compute-resources" => {},
           :"eviction-hard"     => ""
         },
         system_certificates: [],
         dex:                 {
-          connectors: [
-            expected_dex_json(dex_connector_ldap.id, certificate).merge(
-              server:    "ldap_host_#{dex_connector_ldap.id}.com:636",
-              start_tls: false
-            )
-          ]
+          connectors: [connector_json]
         }
       }
-      get :show do
-        expect(json).to eq(expected_json)
-        delete(dex_connector_ldap)
-      end
     end
-  end
 
-  context "with dex LDAP connectors starttls" do
-    it "has dex LDAP connectors" do
-      dex_connector_ldap = create(:dex_connector_ldap, :starttls, :anon_admin)
-      CertificateService.create(service: dex_connector_ldap, certificate: certificate)
-
-      expected_json = {
-        registries:          [],
-        kubelet:             {
-          :"compute-resources" => {},
-          :"eviction-hard"     => ""
-        },
-        system_certificates: [],
-        dex:                 {
-          connectors: [
-            expected_dex_json(dex_connector_ldap.id, certificate).merge(
-              server:    "ldap_host_#{dex_connector_ldap.id}.com:389",
-              start_tls: true,
-              bind:      {
-                anonymous: true
-              }
-            )
-          ]
-        }
-      }
-      get :show do
-        expect(json).to eq(expected_json)
-        delete(dex_connector_ldap)
+    [:tls, :starttls].each do |security|
+      [:regular_admin, :anon_admin].each do |bind_method|
+        context "with #{security}" do
+          it "has dex LDAP connectors for bind as #{bind_method}" do
+            dex_connector_ldap = create(:dex_connector_ldap, security, bind_method)
+            CertificateService.create(service: dex_connector_ldap, certificate: certificate)
+
+            get :show
+            expect(json).to eq(expected_dex_json(dex_connector_ldap, certificate))
+            # destroy should be unecessary, but here it is if that assumtion is invalidated
+            # dex_connector_ldap.destroy
```

Comment:
You can safely remove these lines. No need to worry about that.

Reply:
Removed, and squashed commits.